### ISSUE TITLE
False positive on E1101 when combining str.join with += operator

### ISSUE THREAD

**Issue by themiurge** [PyCQA/pylint#2911]
### Steps to reproduce
1. Run pylint on the following code snippet:

```python
from collections import namedtuple

SomeClass = namedtuple('SomeClass', ['name'])

items = [SomeClass(name='some name')]

some_str = 'Items:\n'
some_str += ', '.join(item.name for item in items)
```

### Current behavior

The `some_str += ...` line gets flagged as:

```
E1101: Instance of 'str' has no 'name' member (no-member)
```

The following (i.e. removing the `+=` operator) works fine:

```python
...
some_str = some_str + ', '.join(item.name for item in items)
```

The following also works fine:

```python
...
item_names = (item.name for item in items)
some_str = 'Items:\n'
some_str += ', '.join(item_names)
```

### Expected behavior

No error.

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.5.2 (default, Nov 12 2018, 13:43:14) 
[GCC 5.4.0 20160609]
```
